Runtime/Agents/ParallelCompositeAgent.cs

```diff
@@ -20,12 +20,13 @@ public override async UniTask Run(CancellationToken token)
             Logger.Log($"Enter {this.name}.Run()");
 
             var tasks = new UniTask[agents.Count];
+            var taskIndex = 0;
 
-            foreach (var (agent, i) in agents.Select((agent, i) => (agent, i)))
+            foreach (var agent in agents.Where(agent => agent != null && agent != this))
             {
                 agent.Logger = Logger;
                 agent.Random = RandomFactory.CreateRandom();
-                tasks[i] = agent.Run(token);
+                tasks[taskIndex++] = agent.Run(token);
```

Comment:
Shouldn't null be appended to the end of the variable 'tasks' for each count when the agent is null or itself? Is this the intended behavior?

Reply:
Thanks, fixed in 66b3b66.
Fix tasks to be stored in a `List` instead of an `Array`.
`UniTask.WhenAll` had an overload of `IEnumerable<UniTask>`.